src/components/chat/chatActions.tsx

```diff
@@ -26,15 +26,18 @@ const ChatActions: React.FC<ChatActionsProps> = ({
   content,
   feedback,
 }) => {
-  // TODO: default to previously sent feedback
-  // TODO: disable feedback when chat is shared
   const [aiState] = useAIState<typeof AI>();
   const aiFeedback = aiState.reaction;
   const [feedbackSent, setFeedbackSent] = React.useState<null | Feedback>(
     feedback ?? aiFeedback ?? null
   );
+  const pathname = usePathname();
+  const onSharedPage = pathname.includes('/share/');
```

Comment:
you send `shared={true}` through to the main component but here use the url to determine if it is shared.  I know it's hard/impossible to pass the shared prop through the streaming messages but still seems disjointed.  not sure of a better option, other than separate chat vs. share ui or maybe context or AIState?

Reply:
@srkirkland ah, the `shared={true}` prop wasn't actually doing anything anymore. it's just in the `chatActions.tsx` that we care and check if we are on ` /share/` so it knows what buttons to render. i think since we have everything in context (AIState) i should just move all the actions and whatnot down into the furthest child i can, and not pass props for the most part, except when a component is reusable UI or separated for readability